library/src/containers/AsyncApi/AsyncApi.tsx

```diff
@@ -1,176 +1,171 @@
-  }
-
-  private async validateSchema(schema: string | any) {
-    if (typeof schema !== 'string') {
-      schema = JSON.stringify(schema);
-    }
-    return await parser.parse(schema);
-  }
-
-  private beautifySchema(schema: AsyncApi): AsyncApi {
-    return beautifier.beautify(schema);
-  }
-
-  private showComponent(
-    showComponent: boolean,
-    component: React.ReactNode,
-  ): React.ReactNode | null {
-    return showComponent ? component : null;
-  }
-
   render() {
     const { theme, config } = this.props;
-    const { validatedSchema, validated, error } = this.state;
+    const { validatedSchema, error } = this.state;
     const concatenatedConfig: ConfigInterface = {
       ...defaultConfig,
       ...config,
       show: {
         ...defaultConfig.show,
-        ...(config && config.show ? config.show : {}),
+        ...(!!config && config.show),
       },
     };
 
     const concatenatedTheme: ThemeInterface = concatenatedConfig.disableDefaultTheme
       ? (theme as ThemeInterface)
       : { ...defaultTheme, ...theme };
 
-    if (!(validatedSchema && validated)) return null;
+    if (!validatedSchema || !Object.keys(validatedSchema).length) {
+      if (!error) {
+        return null;
+      }
+      return (
+        <ThemeProvider theme={concatenatedTheme}>
+          <AsyncApiWrapper>
+            {concatenatedConfig.showErrors && <ErrorComponent error={error} />}
+          </AsyncApiWrapper>
+        </ThemeProvider>
+      );
+    }
+
+    // todo: handle defaultContentType field here
 
     return (
       <ThemeProvider theme={concatenatedTheme}>
         <AsyncApiWrapper>
-          {this.showComponent(
-            concatenatedConfig.showErrors && Boolean(error),
-            <ErrorComponent error={error} />,
+          {concatenatedConfig.showErrors && !!error && (
+            <ErrorComponent error={error} />
           )}
-          {this.showComponent(
-            concatenatedConfig.show.info && Boolean(validatedSchema.info),
+          {concatenatedConfig.show.info && Boolean(validatedSchema.info) && (
             <InfoComponent
               info={validatedSchema.info}
               servers={validatedSchema.servers}
               showServers={
                 concatenatedConfig.show.servers &&
                 Boolean(validatedSchema.servers)
               }
-            />,
-          )}
-          {this.showComponent(
-            concatenatedConfig.show.security &&
-              Boolean(validatedSchema.security),
-            <Security
-              security={validatedSchema.security as SecurityScheme[]}
-            />,
+            />
           )}
-          {this.showComponent(
-            concatenatedConfig.show.topics && Boolean(validatedSchema.topics),
-            <TopicsComponent
-              baseTopic={validatedSchema.baseTopic}
-              topics={validatedSchema.topics}
-            />,
+          {concatenatedConfig.show.security &&
+            validatedSchema.components &&
```

Comment:
I removed `Boolean` in other parts of file because TS doesn't handle it so well in comparision to `!!` syntax (or even plain object.field.field... syntax)

And Boolean did not work in this file before so I will remove it altogether